tests/unit/commands/deploy/test_deploy_list.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 UShareSoft SAS, All rights reserved
+#
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from unittest import TestCase
+
+import pyxb
+from mock import patch
+from uforge.application import Api
+import hammr.commands.deploy
+from hammr.utils import constants
+from uforge.objects.uforge import *
+from uforge.objects import uforge
+
+class TestDeployList(TestCase):
+    # URL: users/{uid}/deployments/{did}/status
+    @patch('uforge.application.Api._Users._Deployments._Status.Getdeploystatus')
+    @patch('uforge.application.Api._Users._Deployments.Getall')
+    @patch('texttable.Texttable.add_row')
+    def test_do_list_gives_correct_number_of_deployments(self, mock_table_add_row, mock_api_deployments_getall, mock_api_getdeploystatus):
+        # given
+        i = hammr.commands.deploy.Deploy()
+        i.api = Api("url", username="username", password="password", headers=None,
+                    disable_ssl_certificate_validation=False, timeout=constants.HTTP_TIMEOUT)
+        i.login = "login"
+        i.password = "password"
+        self.create_deployments(mock_api_deployments_getall)
+        self.prepare_mock_api_getdeploystatus(mock_api_getdeploystatus)
+
+        # when
+        i.do_list("")
```

Comment:
The do_list method must take an argument (it is defined in ussclicore), so the test gives an empty string. I put it to None, it may be more clear